liquibase-service/src/main/resources/db/rdb_modern/routines/023-sp_d_interview_postprocessing-001.sql

```diff
@@ -0,0 +1,525 @@
+
+        BEGIN
+            TRANSACTION;
+        SET
+            @PROC_STEP_NO = @PROC_STEP_NO + 1;
+        SET
+            @PROC_STEP_NAME = 'Remove existing comments from D_INTERVIEW_NOTE';
+
+        /* 
+        There is no update step for D_INTERVIEW_NOTE.
+        This is because each time a new interview note is created or an existing interview note is updated/deleted,
+        all associated interview notes are deleted from NBS_ODSE.dbo.NBS_ANSWER.
+
+        So, it is necessary to delete and reinsert every time. 
+        */
+        DELETE FROM dbo.D_INTERVIEW_NOTE
+        WHERE D_INTERVIEW_KEY IN (SELECT D_INTERVIEW_KEY FROM #INTERVIEW_NOTE_INIT WHERE D_INTERVIEW_KEY IS NOT NULL);
+
+
+        SELECT @RowCount_no = @@ROWCOUNT;
+
+        INSERT INTO [dbo].[job_flow_log]
+        (batch_id, [Dataflow_Name], [package_Name], [Status_Type], [step_number], [step_name], [row_count])
+        VALUES (@batch_id, 'D_INTERVIEW', 'D_INTERVIEW', 'START', @Proc_Step_no, @Proc_Step_Name, @RowCount_no);
+
+
+        COMMIT TRANSACTION;
+
+
+        BEGIN
+            TRANSACTION;
+        SET
+            @PROC_STEP_NO = @PROC_STEP_NO + 1;
+        SET
+            @PROC_STEP_NAME = 'INSERT INTO nrt_interview_note_key';
+
+        INSERT INTO dbo.nrt_interview_note_key(d_interview_key,nbs_answer_uid)
+        SELECT
+            D_INTERVIEW_KEY,
+            NBS_ANSWER_UID
+        FROM #INTERVIEW_NOTE_INIT
+        WHERE D_INTERVIEW_NOTE_KEY IS NULL;
+
+
+        SELECT @RowCount_no = @@ROWCOUNT;
+
+        INSERT INTO [dbo].[job_flow_log]
+        (batch_id, [Dataflow_Name], [package_Name], [Status_Type], [step_number], [step_name], [row_count])
+        VALUES (@batch_id, 'D_INTERVIEW', 'D_INTERVIEW', 'START', @Proc_Step_no, @Proc_Step_Name, @RowCount_no);
+
+
+        COMMIT TRANSACTION;
+
+
+        BEGIN
+            TRANSACTION;
+        SET
+            @PROC_STEP_NO = @PROC_STEP_NO + 1;
+        SET
+            @PROC_STEP_NAME = 'INSERTING INTO D_INTERVIEW_NOTE';
+
+
+        INSERT INTO dbo.D_INTERVIEW_NOTE (D_INTERVIEW_KEY,
+                                          D_INTERVIEW_NOTE_KEY,
+                                          NBS_ANSWER_UID,
+                                          USER_FIRST_NAME,
+                                          USER_LAST_NAME,
+                                          USER_COMMENT,
+                                          COMMENT_DATE)
+        SELECT ixnk.D_INTERVIEW_KEY,
+               ixnk.D_INTERVIEW_NOTE_KEY,
+               ixn.NBS_ANSWER_UID,
+               ixn.USER_FIRST_NAME,
+               ixn.USER_LAST_NAME,
+               ixn.USER_COMMENT,
+               ixn.COMMENT_DATE
+        FROM #INTERVIEW_NOTE_INIT ixn
+            LEFT JOIN dbo.nrt_interview_note_key ixnk
+                ON ixn.D_INTERVIEW_KEY = ixnk.D_INTERVIEW_KEY
+                AND ixn.NBS_ANSWER_UID = ixnk.NBS_ANSWER_UID;
+
+
+        SELECT @RowCount_no = @@ROWCOUNT;
+
+        INSERT INTO [dbo].[job_flow_log]
```

Comment:
LGTM! I would suggested a final log into job_flow_log with STEP value 999 to indicate a COMPLETE run.

Reply:
Done 👍